bridge/ApiRequestHandler.cpp

```diff
@@ -40,358 +18,67 @@ httpd::Response makeErrorResponse(httpd::StatusCode statusCode, const std::strin
-    opus._parameters.emplace_back("minptime", "10");
-    opus._parameters.emplace_back("useinbandfec", "1");
-
-    audioChannel._payloadType.set(opus);
-    audioChannel._rtpHeaderExtensions.emplace_back(1, "urn:ietf:params:rtp-hdrext:ssrc-audio-level");
-    audioChannel._rtpHeaderExtensions.emplace_back(3, "http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time");
-}
+} // namespace
 
-void addDefaultVideoProperties(api::EndpointDescription::Video& videoChannel)
+namespace bridge
 {
-    {
-        api::EndpointDescription::PayloadType vp8;
-        vp8._id = codec::Vp8::payloadType;
-        vp8._name = "VP8";
-        vp8._clockRate = codec::Vp8::sampleRate;
-        vp8._rtcpFeedbacks.emplace_back("goog-remb", utils::Optional<std::string>());
-        vp8._rtcpFeedbacks.emplace_back("nack", utils::Optional<std::string>());
-        vp8._rtcpFeedbacks.emplace_back("nack", utils::Optional<std::string>("pli"));
-        videoChannel._payloadTypes.push_back(vp8);
-    }
-
-    {
-        api::EndpointDescription::PayloadType vp8Rtx;
-        vp8Rtx._id = codec::Vp8::rtxPayloadType;
-        vp8Rtx._name = "rtx";
-        vp8Rtx._clockRate = codec::Vp8::sampleRate;
-        vp8Rtx._parameters.emplace_back("apt", std::to_string(codec::Vp8::payloadType));
-        videoChannel._payloadTypes.push_back(vp8Rtx);
-    }
-
-    videoChannel._rtpHeaderExtensions.emplace_back(3, "http://www.webrtc.org/experiments/rtp-hdrext/abs-send-time");
-    videoChannel._rtpHeaderExtensions.emplace_back(4, "urn:ietf:params:rtp-hdrext:sdes:rtp-stream-id");
-}
 
-bridge::RtpMap makeRtpMap(const api::EndpointDescription::PayloadType& payloadType)
+ApiRequestHandler::ApiRequestHandler(bridge::MixerManager& mixerManager, transport::SslDtls& sslDtls)
+    : ActionContext(mixerManager, sslDtls)
+#if ENABLE_LEGACY_API
+      ,
+      _legacyApiRequestHandler(std::make_unique<LegacyApiRequestHandler>(mixerManager, sslDtls))
+#endif
 {
-    bridge::RtpMap rtpMap;
-
-    if (payloadType._name.compare("opus") == 0)
-    {
-        rtpMap = bridge::RtpMap(bridge::RtpMap::Format::OPUS,
-            payloadType._id,
-            payloadType._clockRate,
-            payloadType._channels);
-    }
-    else if (payloadType._name.compare("VP8") == 0)
-    {
-        rtpMap = bridge::RtpMap(bridge::RtpMap::Format::VP8, payloadType._id, payloadType._clockRate);
-    }
-    else if (payloadType._name.compare("rtx") == 0)
-    {
-        rtpMap = bridge::RtpMap(bridge::RtpMap::Format::VP8RTX, codec::Vp8::rtxPayloadType, codec::Vp8::sampleRate);
-    }
-    else
-    {
-        throw httpd::RequestErrorException(httpd::StatusCode::BAD_REQUEST,
-            utils::format("rtp payload '%s' not supported", payloadType._name.c_str()));
-    }
-
-    for (const auto& parameter : payloadType._parameters)
-    {
-        rtpMap._parameters.emplace(parameter.first, parameter.second);
-    }
-
-    return rtpMap;
+    _actionMap.emplace(ApiActions::ABOUT, ::bridge::handleAbout);
```

Comment:
I wonder. One way to put it is that this looks like C-implementation boiler plate of virtual methods.
Since you already added multiple inheritance to ApiRequestHandler, one could have put all the requesthandlers in super classes that one inherits from ApiRequestHandler and call them directly form onRequest. That way you would have type check, that method is implemented, at compile time. I am not that fond of this interpretation of request URL into enum into "switch/map" into function pointers to implementation.